app/src/main/java/tech/relaycorp/gateway/background/CheckInternetAccess.kt

```diff
@@ -0,0 +1,15 @@
+package tech.relaycorp.gateway.background
+
+import javax.inject.Inject
+
+class CheckInternetAccess
+@Inject constructor(
+    private val pingRemoteServer: PingRemoteServer
+) {
+    suspend fun check() =
+        pingRemoteServer.pingHostname(INTERNET_TEST_WEBSITE)
+
+    companion object {
+        private const val INTERNET_TEST_WEBSITE = "http://google.com"
```

Comment:
Can you please use the HTTPS version? Blocked HTTP websites can return `200 OK` if the censor/ISP intercepts the request. They can't do that over TLS.